src/Predavanje02/Inventory.kt

```diff
@@ -1,65 +1,80 @@
-                    inventory.currentUser.addEmployee(Employee(name, lastName, role!!))
-                }
-                2 -> {
-                    println("Title: ")
-                    val title = readln()
-                    println("Category: 1. Food, 2. Tech, 3. Book")
-                    val category = readln().toInt()
-                    val productCategory = ProductCategory.entries.find { it.value == category }
-                    println("Quantity: ")
-                    val quantity = readln().toInt()
-                    inventory.addToInventory(Product(title, productCategory!!, quantity))
+    loginHelper.performLogin(
+        username = username,
+        onFailure = { exitProcess(0) }) { currentUser ->
+        // Proceed with the program. The current user is available here if needed.
+    }
+    do {
+        println(inventory.currentUser.name)
+        println("Press number: 1. Add Employee, 2. Add Product, 3. Update Stock, 4. Print all products 5. Exit")
+        when (readln().toInt()) {
+            1 -> {
+                println("Name: ")
+                val name = readln()
+                println("Last name: ")
+                val lastName = readln()
+                println(buildString {
+                    Role.entries.forEachIndexed { index, role -> append("${index + 1}.$role ") }
+                })
+                val input = readln().toIntOrNull()
+                val roleIndex = if (input == null || (input - 1) >= Role.entries.count()) {
+                    println("Unsupported operation! Exiting app...")
+                    exitProcess(0)
+                } else {
+                    input - 1
                 }
-                3 -> {
-                    println("Title: ")
-                    val title = readln()
-                    println("1. Increase product stock, 2. Decrease product stock")
-                    val action = readln().toInt()
-                    println("Quantity: ")
-                    val quantity = readln().toInt()
-                    when(action) {
-                        1 -> inventory.updateStock(title, quantity, StockUpdate.ADD)
-                        2 -> {
-                            inventory.updateStock(title, quantity, StockUpdate.REMOVE)
-                        }
-                    }
+                val role = Role.entries[roleIndex]
+                inventory.currentUser.addEmployee(Employee(name, lastName, role))
+            }
+
+            2 -> {
+                println("Title: ")
+                val title = readln()
+                println("Category: 1. Food, 2. Tech, 3. Book")
+                val category = readln().toInt()
+                val productCategory = ProductCategory.entries.find { it.value == category }
+                println("Quantity: ")
+                val quantity = readln().toInt()
+                inventory.addToInventory(Product(title, productCategory!!, quantity))
```

Comment:
Pripazi na not-null assertion operator (!!). Jesi siguran da `productCategory` neće biti null?